scio-core/src/main/scala/com/spotify/scio/experimental/package.scala

```diff
@@ -77,7 +77,12 @@ package object experimental {
         val query = if (newSource != null) newSource else bqt.query.get
         self.bigQuerySelect(query).map(bqt.fromTableRow)
       } else {
-        throw new IllegalArgumentException(s"Missing table or query field in companion")
+        if (newSource != null) {
+          self.bigQuerySelect(newSource).map(bqt.fromTableRow)
```

Comment:
Good point. I updated the docs and behavior so `newSource` can be either table or query if `query` or `table` is missing form companion object.